sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -272,16 +272,32 @@ class Dataset[T] private[sql](
       numRows: Int,
       truncate: Int): Seq[Seq[String]] = {
     val newDf = toDF()
-    val castCols = newDf.logicalPlan.output.map { col =>
-      // Since binary types in top-level schema fields have a specific format to print,
-      // so we do not cast them to strings here.
-      if (col.dataType == BinaryType) {
-        Column(col)
-      } else {
-        Column(col).cast(StringType)
-      }
+    val data = newDf.logicalPlan match {
```

Comment:
to deduplicate code, how about
```
val newDf = logicalPlan match {
  case c: CommandResult =>
    Dataset.ofRows(sparkSession, LocalRelation(c.output, c.rows)
  case _ => toDf()
}
same code as before...
```

Reply:
if people exclude `ConvertToLocalRelation`, then we shouldn't do local evaluation here as well.

Reply:
All right, then https://github.com/apache/spark/pull/40779/commits/d8210ff849727961681b7d9df71b70e19ae24d41 contains the deduplication. Thanks for the idea.